service/r2/routes.go

```diff
@@ -64,6 +64,27 @@ func createNamespace(s *service, r *http.Request) (data interface{}, err error)
 	return newNamespaceJSON(view), nil
 }
 
+func validateNamespace(s *service, r *http.Request) (data interface{}, err error) {
+	vars := mux.Vars(r)
+	var rsj ruleSetJSON
+	if err := parseRequest(&rsj, r.Body); err != nil {
+		return nil, err
+	}
+	if vars[namespaceIDVar] != rsj.Namespace {
+		return nil, fmt.Errorf(
+			"NamespaceID param: %s and RuleSet namespaceID: %s do not match",
+			vars[namespaceIDVar],
+			rsj.Namespace,
+		)
+	}
+
+	rss := rsj.ruleSetSnapshot()
```

Comment:
NIT:
Given that this function is only called in this file, and ruleSetSnapshot would be the only stuct with methods on it. Perhaps make this function take a ruleSetJSON and return a ruleSetSnapshot. The It might also have it as sort of public util function.